### PR TITLE
[14.0] [mig] pos_reset_search

### PR DESCRIPTION
**Pull request by dsolanki-initos**

### CODE REVIEW

**Comment by dsolanki-initos** [comment]
It would be great if we get the reviews here

**Comment by dsolanki-initos** [comment]
@flotho It would be great if you could add the reviews

**Comment by dsolanki-initos** [comment]
It would be great if we could get the reviews here

**Comment by dsolanki-initos** [comment]
It would be great if we could get the reviews here

**Comment by dsolanki-initos** [comment]
It would be great if we could get the reviews here

**Comment by dsolanki-initos** [comment]
Hi @Fkawala It would be nice if you can review this PR

**Comment by Fkawala** [comment]
@dsolanki-initos sure will do that ASAP (probably tomorrow)

**Comment by dsolanki-initos** [comment]
Hi @ivantodorovich It would be nice if you can review the PR

File: pos_reset_search/static/src/js/reset.js [hunk 1]
```diff
@@ -1,12 +1,39 @@
 odoo.define("product_click_search_reset", function (require) {
```
Comment (ivantodorovich) [inline pos_reset_search/static/src/js/reset.js]:
```suggestion
odoo.define("pos_reset_search.ProductScreen", function (require) {
```

Also rename file to `ProductScreen.js` please

File: pos_reset_search/static/src/js/reset.js [hunk 2]
```diff
@@ -1,12 +1,39 @@
...
+                if (ProductWidget) {
+                    const ProductWidgetChildrens = ProductWidget.__owl__.children;
+                    const ProductsWidgetControlPanel = _.find(
+                        ProductWidgetChildrens,
+                        function (children) {
+                            if (children.el.className === "products-widget-control") {
+                                return children;
+                            }
+                        }
+                    );
```
Comment (ivantodorovich) [inline pos_reset_search/static/src/js/reset.js]:
```suggestion
                    const ProductsWidgetControlPanel = ProductWidgetChildren.find(
                        child => child.el.className === "products-widget-control"
                    );
```

Also "children" is already plural (of "child"), so you can remove the "s"
Please note there are other occurrences like this one

File: pos_reset_search/static/src/js/reset.js [hunk 3]
```diff
@@ -1,12 +1,39 @@
...
+        class extends ProductScreen {
+            async _clickProduct(event) {
+                const ProductScreenChildrens = this.__owl__.children;
+                const ProductWidget = _.find(ProductScreenChildrens, function (
+                    children
+                ) {
+                    if (children.el.className === "products-widget") {
+                        return children;
+                    }
+                });
```
Comment (ivantodorovich) [inline pos_reset_search/static/src/js/reset.js]:
```suggestion
                const ProductWidget = ProductScreenChildren.find(
                    child => child.el.className === "products-widget"
                );
```

File: pos_reset_search/static/src/js/reset.js [hunk 4]
```diff
@@ -1,12 +1,39 @@
...
-        },
-    });
+
+    const ProductScreen = require("point_of_sale.ProductScreen");
+    const Registries = require("point_of_sale.Registries");
+
+    const PosProductScreen = (ProductScreen) =>
+        class extends ProductScreen {
+            async _clickProduct(event) {
+                const ProductScreenChildrens = this.__owl__.children;
```
Comment (ivantodorovich) [inline pos_reset_search/static/src/js/reset.js]: I'm not so familiar with owl yet to know if this is ok, but I trust you here :)

File: pos_reset_search/readme/CONTRIBUTORS.rst [hunk 5]
```diff
@@ -0,0 +1,4 @@
+- Le Nid
+- CoopITEasy
+
+* Dhara Solanki <dhara.solanki@initos.com>
```
Comment (ivantodorovich) [inline pos_reset_search/readme/CONTRIBUTORS.rst]:
```suggestion
* Le Nid
* CoopITEasy
* Dhara Solanki <dhara.solanki@initos.com>
```

Would you please normalize this too?

**Review by ivantodorovich — CHANGES_REQUESTED** [review]

**Review by fshah-initos — APPROVED** [review]
LGM. Code review and Functional tests.

**Review by fshah-initos — APPROVED** [review]
:+1:

File: pos_reset_search/static/src/js/reset.js (hunk 1 above)
Comment (ivantodorovich) [reply pos_reset_search/static/src/js/reset.js]: This one's still pending :)

We can merge after this is AFAICS

**Comment by dsolanki-initos** [comment]
Hello, @ivantodorovich Changes are done can you please re-review it?

File: pos_reset_search/static/src/js/reset.js (hunk 1 above)
Comment (ivantodorovich) [reply pos_reset_search/static/src/js/reset.js]: This one's still pending @dsolanki-initos

**Comment by hkapatel-initos** [comment]
Hello, @ivantodorovich Changes are done can you please re-review it?

**Review by ivantodorovich — APPROVED** [review]

**Comment by hkapatel-initos** [comment]
Hi, @ivantodorovich can we try to merge this again?

**Pull request merged**